packages/shared/components/inputs/AssetAmountInput2.svelte

```diff
@@ -0,0 +1,105 @@
+<script lang="ts">
+    import { formatCurrency, localize, parseCurrency } from '@core/i18n'
+    import { getMaxDecimalsFromTokenMetadata } from '@core/token/utils'
+    import {
+        IAsset,
+        TokenStandard,
+        convertToRawAmount,
+        formatTokenAmountDefault,
+        visibleSelectedAccountAssets,
+    } from '@core/wallet'
+    import Big from 'big.js'
+    import { AmountInput, FontWeight, InputContainer, Text } from '@ui'
+    import { getMarketAmountFromAssetValue } from '@core/market/utils'
+
+    export let disabled = false
+    export let asset: IAsset | undefined = $visibleSelectedAccountAssets?.baseCoin
+    export let rawAmount: string | undefined = undefined
+    export let unit: string | undefined = undefined
+    export let amount: string | undefined =
```

Comment:
perhaps opportunity to refactor and clean up a bit. a biit confusing what amount vs rawamount is here